src/hooks/program.ts

```diff
@@ -570,175 +571,75 @@ export const useProgramPlansEnrollmentsAggregateList = (programPlanIds: string[]
-            listPrice: data.program_content_by_pk.list_price,
-            salePrice: data.program_content_by_pk.sale_price,
-            soldAt: data.program_content_by_pk.sold_at && new Date(data.program_content_by_pk.sold_at),
-            contentBodyId: data.program_content_by_pk.program_content_body?.id,
-            programContentBody: data.program_content_by_pk.program_content_body
-              ? {
-                  id: data.program_content_by_pk.program_content_body.id,
-                  type: data.program_content_by_pk.program_content_body.type || null,
-                  description: data.program_content_by_pk.program_content_body.description || '',
-                  data: data.program_content_by_pk.program_content_body.data,
-                }
-              : null,
-            videos: data.program_content_by_pk.program_content_videos.map(v => ({
-              id: v.attachment.id,
-              size: v.attachment.size,
-              options: v.attachment.options,
-              data: v.attachment.data,
-            })),
-            audios: data.program_content_by_pk.program_content_audios.map(v => ({
-              data: v.data,
-            })),
-            attachments: data.program_content_by_pk.program_content_attachments.map(u => ({
-              id: u.attachment_id,
-              data: u.data,
-              options: u.options,
-              createdAt: u.created_at,
-            })),
-            lastProgress: data.program_content_by_pk.program_content_progress[0]?.last_progress || 0,
-          },
-    [data],
-  )
-
-  return {
-    loadingProgramContent: loading,
-    errorProgramContent: error,
-    programContent,
-    refetchProgramContent: refetch,
-  }
-}
+        isEquity: boolean
+      }
+  >()
 
-export const useEnrolledProgramIds = (memberId: string) => {
-  const { loading, error, data, refetch } = useQuery<
-    hasura.GET_ENROLLED_PROGRAMS,
-    hasura.GET_ENROLLED_PROGRAMSVariables
-  >(
-    gql`
-      query GET_ENROLLED_PROGRAMS($memberId: String!) {
-        program_enrollment(where: { member_id: { _eq: $memberId } }, distinct_on: program_id) {
-          program_id
-        }
-        program_plan_enrollment(where: { member_id: { _eq: $memberId } }) {
-          program_plan {
-            id
-            program_id
+  useEffect(() => {
+    // TODO: Axios staring from v0.22.0 CancelToken deprecated
+    const CancelToken = axios.CancelToken
+    const source = CancelToken.source()
+    if (!authToken && contentId) {
+      setLoadingProgramContent(true)
+      const route = `/programs/${programId}/contents/${contentId}/trial`
+      const getProgramContent = async () => {
+        try {
+          const { data } = await axios.get(`${process.env.REACT_APP_LODESTAR_SERVER_ENDPOINT}${route}`, {
+            cancelToken: source.token,
+          })
+          setLoadingProgramContent(false)
+          setProgramContent(data)
```

Comment:
need to add a conversion function here,  directly use API response data will cause debugging hardly, and coupling with the specifically lodestar-server handler

Reply:
I don't understand why it is necessary to add a conversion function, and why directly using the API to respond to data will cause debugging difficulties?
Or do you mean that errors should be caught and error status returned?

Reply:
After the meeting, the conclusion is that I understand what you are trying to say, but I would like to discuss it with the team at the code review.